server/src/main/java/org/elasticsearch/index/fielddata/plain/SortedNanosecondsNumericSortField.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.index.fielddata.plain;
+
+import org.apache.lucene.search.SortField;
+import org.apache.lucene.search.SortedNumericSortField;
+
+public class SortedNanosecondsNumericSortField extends SortedNumericSortField {
+
+    private final SortField field;
+
+    public SortedNanosecondsNumericSortField(SortedNumericSortField field) {
```

Comment:
Can we do:
```
public SortedNanosecondsNumericSortField(String fieldName, SortedNumericSelector.Type selectorType) {
      super(fieldName, SortField.Type.LONG, selectorType);
}

public SortedNanosecondsNumericSortField(String fieldName) {
      super(fieldName, SortField.Type.LONG);
}
// ... plus the ones that expose the reverse option
```
to avoid the nested `field` member ?

Reply:
done